tasks/insights-client.yml

```diff
@@ -36,6 +36,59 @@
     insertafter: "#auto_update"
     line: auto_update={{ rhc_insights.autoupdate | d(true) | bool }}
 
+- name: Check ansible host in insights-client config
+  when:
+    - rhc_insights.ansible_host is defined
+    - not rhc_insights.ansible_host is none
+    - rhc_insights.ansible_host != omit
+  lineinfile:
+    path: "{{ __rhc_insights_conf }}"
+    regexp: "^ansible_host="
+    state: "{{ 'present' if rhc_insights.ansible_host != __rhc_state_absent else 'absent' }}"
+    line: "{{ ('ansible_host='+rhc_insights.ansible_host) if rhc_insights.ansible_host != __rhc_state_absent }}"
```

Comment:
sounds like you could rather add
```yaml
  - rhc_insights.ansible_host != __rhc_state_absent
```
as 3rd condition in the `when:` of this task: you don't need to check what exists if you need to drop it altogether